deltametrics/section.py

```diff
@@ -518,10 +703,268 @@ def __init__(self, x=-1):
+        value is given, the `azimuth` defaults to ``90``.
+
+    origin : :obj:`tuple` or `list` of `int`, optional
+        The `origin` of the radial section. This is the "start" of the radial
+        line. If no value is given, the origin defaults to the center of the
+        x-direction of the model domain, and offsets into the domain a
+        distance of ``y == L0``, if these values can be determined. I.e., the
+        origin defaults to be centered over the channel inlet. If no value is
+        given and these values cannot be determined, the origin defaults to
+        ``(0, 0)``.
+
+    length : :obj:`float`, `int`, optional
+        The length of the section (note this must be given in pixel length).
+        If no value is given, the length defaults to the length required to
+        reach a model boundary (if a connection to underlying `Cube` exists).
+        Otherwise, length is set to ``1``.
+
+    **kwargs
+        Keyword arguments are passed to `BaseSection.__init__()`. Supported
+        options are `name`.
+
+    Returns
+    -------
+    section : :obj:`RadialSection`
+        `RadialSection` object with specified parameters. The section is
+        automatically connected to the underlying `Cube` data source if the
+        :obj:`register_section` method of a `Cube` is used to set up the
+        section, or the `Cube` is passed as the first positional argument
+        during instantiation.
+
+    Examples
+    --------
+
+    To create a `RadialSection` that is registered to a `DataCube` at
+    specified `origin` coordinate, and spans the entire model domain:
+
+    .. plot::
+        :include-source:
+
+        >>> rcm8cube = dm.sample_data.cube.rcm8()
+        >>> rcm8cube.register_section('radial', dm.section.RadialSection(azimuth=45))
+        >>>
+        >>> # show the location and the "velocity" variable
+        >>> fig, ax = plt.subplots(2, 1, figsize=(8, 4))
+        >>> rcm8cube.show_plan('eta', t=-1, ax=ax[0], ticks=True)
+        >>> rcm8cube.sections['radial'].show_trace('r--', ax=ax[0])
+        >>> rcm8cube.sections['radial'].show('velocity', ax=ax[1])
+        >>> plt.show()
     """
+    def __init__(self, *args, azimuth=None, origin=None, length=None,
+                 **kwargs):
+        self._input_azimuth = azimuth
+        self._input_origin = origin
+        self._input_length = length
+        super().__init__('radial', *args, **kwargs)
 
-    def __init__(self, apex, radius):
-        raise NotImplementedError
+    def _compute_section_coords(self):
+
+        # determine the azimuth
+        if (self._input_azimuth is None):
+            self.azimuth = 90
+        else:
+            self.azimuth = self._input_azimuth
+
+        # determine the origin of the line
+        if (self._input_origin is None):
+            if not (self.cube is None):
+                land_width = utils.guess_land_width_from_land(
+                    self.cube['eta'][-1, :, 0])
```

Comment:
I think this should be consistent with the method used in `CircularSection`. Right now this implementation is putting the default origin on the opposite side of the domain when applied to a `StratigraphyCube` made from the example data. (Implementation for `CircularSection` is working as expected)

Code snippet if you want to see what I mean:
```
sc8cube = dm.cube.StratigraphyCube.from_DataCube(rcm8cube)
sc8cube.register_section('radial', dm.section.RadialSection(azimuth=45))
# show the location and the "velocity" variable
fig, ax = plt.subplots(2, 1, figsize=(8, 4))
rcm8cube.show_plan('eta', t=-1, ax=ax[0], ticks=True)
sc8cube.sections['radial'].show_trace('r--', ax=ax[0])
sc8cube.sections['radial'].show('velocity', ax=ax[1])
plt.show()
```

Reply:
thanks, good catch. Ultimately, this will be reworked after the netcdf file is updated to include `L0`